src/test/test-copy.c

```diff
@@ -198,20 +198,31 @@ static void test_copy_bytes_regular_file(const char *src, bool try_reflink) {
         fd3 = mkostemp_safe(fn3, O_WRONLY);
         assert_se(fd3 >= 0);
 
-        r = copy_bytes(fd, fd2, (uint64_t) -1, try_reflink);
-        assert_se(r == 0);
+        r = copy_bytes(fd, fd2, max_bytes, try_reflink);
+        if (max_bytes == (uint64_t) -1)
```

Comment:
confusion between (uint64_t) and (size_t)... You shouldn't mix them, otherwise things will explode on 32bit systems...